src/newtondynamics_physics/SysNewton.cpp

```diff
@@ -76,75 +70,114 @@ using osp::Matrix3;
 using osp::Matrix4;
 using osp::Vector3;
 
-namespace
-{
 
 // Callback called for dynamic rigid bodies for applying force and torque
-static void cb_force_torque(const NewtonBody* pBody, dFloat timestep, int threadIndex)
+void SysNewton::cb_force_torque(const NewtonBody* pBody, dFloat timestep, int threadIndex)
```

Comment:
I recommend making a typedef to repreent the thread index type.

It can just be "using ThreadIndex_t = intl" for all i care, just always bugs me to see "int" is all.